cmd/octopus/main.go

```diff
@@ -22,15 +23,24 @@ func main() {
 	// if there is a missing or invalid .env file anywhere, we don't care, just ignore it
 	_ = godotenv.Load()
 
-	clientFactory, err := apiclient.NewClientFactoryFromEnvironment()
+	// initialize our wrapper around survey, which is also used as a flag for whether
+	// we are in interactive mode or automation mode
+	askProvider := question.NewAskProvider(survey.AskOne)
+	_, ci := os.LookupEnv("CI")
+	// TODO move this to some other function and have it look for GITHUB_ACTIONS etc as we learn more about it
+	if ci {
+		askProvider.SetPromptDisabled()
+	}
```

Comment:
I believe we'll need to create a version of the [AutomationEnvironmentProvider](https://github.com/OctopusDeploy/OctopusClients/blob/master/source/Octopus.Server.Client/AutomationEnvironments/AutomationEnvironmentProvider.cs) available in the C# API client. This should provide a more reliable detection mechanism since `CI` isn't honoured by every automation environment.

Reply:
Thanks for the pointer to the existing code, I wasn't aware of that. Agree; the TODO indicates where we'd call it here